store/boltdb/boltdb.go

```diff
@@ -23,15 +24,28 @@ var (
 	ErrBoltBucketOptionMissing = errors.New("boltBucket config option missing")
 )
 
+const (
+	dbRead  os.FileMode = 0444
+	dbWrite os.FileMode = 0644
+)
+
 //BoltDB type implements the Store interface
 type BoltDB struct {
 	client     *bolt.DB
 	boltBucket []byte
 	dbIndex    uint64
+	path       string
+	timeout    time.Duration
+	// TransientConnection should be  set if the application wants to
+	// open more  than connection to the  Bolt DB. This forces an open
+	// and close for every opertion rather than opening the connection
+	// during New
+	TransientConnection bool
```

Comment:
this is going to be a recurring theme when it comes to holding a lock on boltdb and there is absolutely no point in making all the clients aware of this limitation. If the client really look for a dedicated connection, then can opt-in.
IMO, we should not force this on the clients.